railties/lib/rails/backtrace_cleaner.rb

```diff
@@ -23,5 +23,11 @@ def initialize
       end
       add_silencer { |line| !APP_DIRS_PATTERN.match?(line) }
     end
+
+    attr_reader :root
+
+    def root=(path)
+      @root = "#{path.to_s.chomp("/")}/"
+    end
```

Comment:
Yeah, also not liking that we're exposing `root`. (Both the reader/writer should be marked `# :nodoc:` by the way)